diff --git a/rofi-power-menu b/rofi-power-menu
index 9484f68..99bdd95 100755
--- a/rofi-power-menu
+++ b/rofi-power-menu
@@ -16,7 +16,7 @@ set -u
 all=(shutdown reboot suspend hibernate logout lockscreen)
 
 # By default, show all (i.e., just copy the array)
-show=("${all[@]}")
+show=(suspend shutdown reboot lockscreen logout)
 
 declare -A texts
 texts[lockscreen]="lock screen"
@@ -31,7 +31,7 @@ declare -A icons
 icons[lockscreen]="\Uf033e"
 icons[switchuser]="\Uf0019"
 icons[logout]="\Uf0343"
-icons[suspend]="\Uf04b2"
+icons[suspend]="\U23fe"
 icons[hibernate]="\Uf02ca"
 icons[reboot]="\Uf0709"
 icons[shutdown]="\Uf0425"
@@ -40,7 +40,7 @@ icons[cancel]="\Uf0156"
 declare -A actions
 actions[lockscreen]="loginctl lock-session ${XDG_SESSION_ID-}"
 #actions[switchuser]="???"
-actions[logout]="loginctl terminate-session ${XDG_SESSION_ID-}"
+actions[logout]="; echo 'awesome.quit()' | awesome-client"
 actions[suspend]="systemctl suspend"
 actions[hibernate]="systemctl hibernate"
 actions[reboot]="systemctl reboot"
@@ -53,6 +53,7 @@ confirmations=(reboot shutdown logout)
 dryrun=false
 showsymbols=true
 showtext=true
+eng=false
 
 function check_valid {
     option="$1"
@@ -68,7 +69,7 @@ function check_valid {
 }
 
 # Parse command-line options
-parsed=$(getopt --options=h --longoptions=help,dry-run,confirm:,choices:,choose:,symbols,no-symbols,text,no-text,symbols-font: --name "$0" -- "$@")
+parsed=$(getopt --options=h --longoptions=help,eng,dry-run,confirm:,choices:,choose:,symbols,no-symbols,text,no-text,symbols-font: --name "$0" -- "$@")
 if [ $? -ne 0 ]; then
     echo 'Terminating...' >&2
     exit 1
@@ -88,6 +89,7 @@ while true; do
             echo "  rofi -show menu -modi \"menu:rofi-power-menu --choices=shutdown/reboot\""
             echo
             echo "Available options:"
+            echo "  --eng                Use english version text instead of polish"
             echo "  --dry-run            Don't perform the selected action but print it to stderr."
             echo "  --choices CHOICES    Show only the selected choices in the given order. Use /"
             echo "                       as the separator. Available choices are lockscreen,"
@@ -115,6 +117,10 @@ while true; do
             echo "  -h,--help            Show this help text."
             exit 0
             ;;
+        "--eng")
+            eng=true
+            shift 1
+            ;;
         "--dry-run")
             dryrun=true
             shift 1
@@ -166,6 +172,15 @@ while true; do
     esac
 done
 
+if [ "$eng" = "false" ]
+then
+    texts[lockscreen]="zablokuj"
+    texts[logout]="wyloguj"
+    texts[suspend]="uśpij"
+    texts[reboot]="uruchom ponownie"
+    texts[shutdown]="wyłącz"
+fi
+
 if [ "$showsymbols" = "false" -a "$showtext" = "false" ]
 then
     echo "Invalid options: cannot have --no-symbols and --no-text enabled at the same time." >&2

